WireAPI/Sources/WireAPI/Public/HTTP Client/HTTPRequest.swift

```diff
@@ -0,0 +1,73 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+/// A http request to be sent to a server.
+
+public struct HTTPRequest: Equatable {
+
+    /// The path of the endpoint.
+
+    public var path: String
+
+    /// The http method.
+
+    public var method: Method
+
+    /// The request payload data.
+
+    public var body: Data?
+
+    /// The type of the body payload.
+
+    public var contentType: String?
+
+    /// The type of the response payload.
+
+    public var acceptType: String?
```

Comment:
Should we have explicit fields for these values or can it be already aligned with `URLSession`?
`var allHTTPHeaderFields: [String : String?] { get set }`

Reply:
My aim for now was to make it easier to bridge over `ZMTransportRequest` which has these explicit fields. These aren't being used at the moment so I'll delete it.